### PR TITLE
fix(executor): streaming token usage always zero for OpenAI-compatible providers

### LINKED ISSUE #3076: Streaming token usage always zero with OpenAI-compatible providers (DeepSeek)
**Is it a request payload issue?**
[x] No, it's another issue.

---

**Describe the bug**

When using DeepSeek (or other OpenAI-compatible providers) in streaming mode, the usage statistics always show 0 tokens for all fields (input/output/reasoning/cached/total tokens). Latency is also incorrect — the record appears within ~1 second of the request, even while the stream is still in progress. Non-streaming mode works fine.

**Root cause**: When CPA injects `stream_options.include_usage: true` into the request, the upstream provider includes a `usage` object in **every** streaming chunk (with zero/null values in non-terminal chunks, and real values only in the final chunk alongside `finish_reason: "stop"`).

In `ParseOpenAIStreamUsage()`, the `usage` field is extracted from **any** chunk without checking whether it's the terminal chunk. The first chunk with `usage` (all zero) triggers `UsageReporter.Publish()`, and since `UsageReporter` uses `sync.Once`, the real token counts from the final chunk are silently discarded.

This affects both `openai_compat_executor.go` and `kimi_executor.go` (both call `ParseOpenAIStreamUsage`).

**Evidence** — DeepSeek API raw responses:

Non-streaming (usage at top level, correct):
```json
{"id":"...","object":"chat.completion","usage":{"prompt_tokens":10,"completion_tokens":64,"total_tokens":74,...}}
```

Streaming — intermediate chunk (no `usage` OR `usage: null`):
```json
data: {"choices":[{"delta":{"content":"..."},"finish_reason":null}]}
```

Streaming — final chunk (usage **only** here, alongside `finish_reason: "stop"`):
```json
data: {"choices":[{"delta":{"content":""},"finish_reason":"stop"}],"usage":{"prompt_tokens":10,"completion_tokens":39,"total_tokens":49,...}}
```

The fix is to check `choices.0.finish_reason` in `ParseOpenAIStreamUsage()` before accepting `usage`, so only terminal chunks contribute to token counts. This mirrors the existing `FilterSSEUsageMetadata()` logic used for Gemini providers.

---

**CLI Type**
openai-compatibility

**Model Name**
deepseek-v4-pro (applies to any OpenAI-compatible provider supporting `stream_options.include_usage`)

**LLM Client**
claude code

**Expected behavior**

Streaming usage statistics should show actual token counts (matching the final chunk's `usage` values), and latency should reflect the full streaming duration.

---

**OS Type**
- OS: Docker amd64

### PR DESCRIPTION
**Pull request by deshark42**
Fixes #3076

## Problem
When `stream_options.include_usage` is enabled, OpenAI-compatible providers include a `usage` object in every streaming chunk — zero/null values in intermediate chunks, real values only in the final chunk alongside `finish_reason: "stop"`.

`ParseOpenAIStreamUsage()` extracted `usage` from any chunk without checking whether it's terminal. The first (zero-value) chunk triggered `UsageReporter.Publish()`, and since `UsageReporter` uses `sync.Once`, the real usage from the final chunk was silently discarded. All token counts showed 0.

This affected both `openai_compat_executor` and `kimi_executor`.

## Fix
Added a `choices.0.finish_reason` check in `ParseOpenAIStreamUsage()`. Usage is now only accepted when `finish_reason` is present and non-empty (i.e. terminal chunks only). This mirrors the existing `FilterSSEUsageMetadata()` pattern used for Gemini providers.

## Validation
- [x] DeepSeek streaming request shows correct token counts and latency
- [x] Non-streaming usage unchanged

### CODE REVIEW

File: internal/runtime/executor/helps/usage_helpers.go [hunk 1]
```diff
@@ -284,6 +284,14 @@ func ParseOpenAIStreamUsage(line []byte) (usage.Detail, bool) {
...
 		return usage.Detail{}, false
 	}
+	// Only accept usage from terminal chunks (finish_reason set and non-empty).
+	// When stream_options.include_usage is enabled, providers may include a
+	// zero-value usage in every chunk; the real values arrive only in the
+	// final chunk alongside a non-null finish_reason.
+	finishReason := gjson.GetBytes(payload, "choices.0.finish_reason")
+	if !finishReason.Exists() || finishReason.String() == "" {
+		return usage.Detail{}, false
+	}
```
[the remark this answers is not in the document]
Comment (deshark42) [reply internal/runtime/executor/helps/usage_helpers.go]: Verified and applied in 6838d648. The `finish_reason` check was fragile — the "late usage after finish reason" test case in `openai_openai-responses_response_test.go:44` confirms usage can arrive in a trailing chunk with `"choices":[]`, where `choices.0.finish_reason` would not exist. Switched to non-zero token check (`total_tokens == 0 && prompt_tokens == 0`) which handles both the bundled-usage and separate-usage-chunk cases.

File: internal/runtime/executor/helps/usage_helpers.go [hunk 2]
```diff
@@ -284,6 +284,14 @@ func ParseOpenAIStreamUsage(line []byte) (usage.Detail, bool) {
 	if !usageNode.Exists() {
 		return usage.Detail{}, false
 	}
+	// Only accept usage from terminal chunks (finish_reason set and non-empty).
+	// When stream_options.include_usage is enabled, providers may include a
+	// zero-value usage in every chunk; the real values arrive only in the
+	// final chunk alongside a non-null finish_reason.
+	finishReason := gjson.GetBytes(payload, "choices.0.finish_reason")
+	if !finishReason.Exists() || finishReason.String() == "" {
+		return usage.Detail{}, false
```
[the remark this answers is not in the document]
Comment (deshark42) [reply internal/runtime/executor/helps/usage_helpers.go]: Confirmed. The test case "late usage after finish reason" at `openai_openai-responses_response_test.go:44` demonstrates this exact scenario — usage arriving after finish_reason with `"choices":[]`. Fixed in 6838d648 by checking for non-zero token counts instead of `choices.0.finish_reason`, which works regardless of whether usage arrives bundled with the stop chunk or in a separate trailing chunk.

**Comment by deshark42** [comment]
Addressed review feedback:
- Replaced `finish_reason` check with non-zero token check (6838d648)
- Shortened comment to one line (ccf931ff)

**Review by luispater — CHANGES_REQUESTED** [review]
### Summary
Fixes a streaming accounting issue for OpenAI-compatible providers when `stream_options.include_usage` is enabled.

### Blocking
- This PR is currently not mergeable (`mergeable: CONFLICTING`, `mergeStateStatus: DIRTY`). Please rebase onto `dev` and resolve conflicts.

### Non-blocking
- The PR body mentions a `finish_reason` check, but the final implementation uses a non-zero token heuristic; please update the description.
- Consider adding a focused unit test for `ParseOpenAIStreamUsage()` covering `usage: null` intermediate chunks and the final usage chunk.

### Test plan
- Not run locally.
